lib/rubocop/cop/style/redundant_self.rb

```diff
@@ -43,6 +43,7 @@ module Style
       #   end
       class RedundantSelf < Cop
         MSG = 'Redundant `self` detected.'.freeze
+        KERNEL_METHODS = Kernel.methods(false)
```

Comment:
The problem with this approach is that the methods you'll get might not match the target Ruby version (e.g. if you're running RuboCop with a different version of Ruby).

Reply:
Good point. Do we need to maintain a list for each version? I’ll try to think of something simpler.

Reply:
I'd rather avoid that. I was just pointing out that this solution is not completely reliable.